src/main/java/com/google/api/codegen/configgen/NodeFinder.java

```diff
@@ -15,12 +15,45 @@
 package com.google.api.codegen.configgen;
 
 import com.google.api.codegen.configgen.nodes.ConfigNode;
+import com.google.api.codegen.configgen.nodes.ListItemConfigNode;
+import com.google.api.codegen.configgen.nodes.NullConfigNode;
 import com.google.common.collect.Iterables;
 import java.util.Iterator;
 import java.util.NoSuchElementException;
 
 /** Utilities for finding and traversing ConfigNodes. */
 public class NodeFinder {
+  public static ConfigNode findByValue(ConfigNode parentNode, String value) {
+    for (ConfigNode childNode : getChildren(parentNode)) {
+      if (value.equals(childNode.getText())) {
+        return childNode;
+      }
+    }
+
+    return new NullConfigNode();
+  }
+
+  public static boolean hasChild(ConfigNode parentNode) {
```

Comment:
nit: Comment this method, and maybe consider changing the title. From a readability perspective, it's hard to tell when browsing the code how `hasChild(node)` should be different from `node.getChild().isPresent()`. What does "having a child" mean that requires so many checks?